### PR TITLE
Implement Blur Detection on Doc Capture 🌻 

### PR DESCRIPTION
**Pull request by jumaallan**
### **User description**
Story: https://app.shortcut.com/smileid/story/xxx

## Summary

A few sentences/bullet points about the changes

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

___

### **PR Type**
Enhancement

___

### **Description**
• Add blur detection algorithm for document capture quality
• Implement adaptive sampling based on image resolution
• Add new blur directive with user feedback message
• Update changelog for version 11.0.5 release

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>Util.kt</strong><dd><code>Implement blur detection algorithm with adaptive sampling</code></dd></summary>
<hr>

lib/src/main/java/com/smileidentity/util/Util.kt

• Add <code>calculateBlur()</code> function with adaptive sampling algorithm<br> • <br>Import math utilities (<code>min</code>, <code>sqrt</code>) for blur calculations<br> • Implement <br>luminance gradient analysis for blur detection<br> • Add comprehensive <br>error handling and logging

</details>

  </td>
  <td><a href="https://github.com/smileidentity/android/pull/651/files#diff-268190d5b630578aa6ef510ff7bd104bf0c91f8c6033a31eae162417c9061995">+87/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>DocumentCaptureViewModel.kt</strong><dd><code>Integrate blur detection into document capture flow</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

lib/src/main/java/com/smileidentity/viewmodel/document/DocumentCaptureViewModel.kt

• Add <code>BLUR_THRESHOLD</code> constant (10.0) for blur detection<br> • Integrate <br>blur detection in image analysis pipeline<br> • Add <code>BlurryDocument</code> <br>directive enum value<br> • Update UI state when blur threshold not met

</details>

  </td>
  <td><a href="https://github.com/smileidentity/android/pull/651/files#diff-0f38750558e9a7901f4af936b460a5b992af5d6cfed9a390b7e3868c937e1d7f">+14/-1</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>strings.xml</strong><dd><code>Add blur detection user message string</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

lib/src/main/res/values/strings.xml

• Add <code>si_doc_v_capture_directive_blurry_document</code> string resource<br> • <br>Provide user-facing message for blurry document detection

</details>

  </td>
  <td><a href="https://github.com/smileidentity/android/pull/651/files#diff-2a7fd72ed64345780392ca16d959ede38bc4180bacb0ac1c6b19234cd5b94fef">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Update changelog for blur detection feature</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

CHANGELOG.md

• Add version 11.0.5 section for unreleased changes<br> • Document blur <br>detection feature addition

</details>

  </td>
  <td><a href="https://github.com/smileidentity/android/pull/651/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about PR-Agent usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -1,5 +1,40 @@
 <component name="ProjectCodeStyleConfiguration">
   <code_scheme name="Project" version="173">
+    <JavaCodeStyleSettings>
```
Comment (robin-smileid) [inline .idea/codeStyles/Project.xml:3]: Any particular reason why we change the code styles with this PR?

File: lib/VERSION [hunk 2]
```diff
@@ -1 +1 @@
-11.0.5-SNAPSHOT
\ No newline at end of file
+11.0.4-SNAPSHOT
```
Comment (robin-smileid) [inline lib/VERSION:1]: We need to remove this change if we release 11.0.5.

File: lib/src/main/java/com/smileidentity/util/Util.kt [hunk 3]
```diff
@@ -114,6 +116,91 @@ fun calculateLuminance(imageProxy: ImageProxy): Double {
...
+ * Calculates the blur level of an image using luminance gradients with adaptive sampling.
+ * Higher values indicate sharper images (less blur). Lower values indicate more blur.
+ *
+ * @param imageProxy The camera image to analyze
+ * @return Average gradient value (0.0 = maximum blur, higher values = sharper image)
+ */
+fun calculateBlur(imageProxy: ImageProxy): Float {
+    // Return 0.0 (maximum blur) for invalid inputs
+    if (imageProxy.planes.isEmpty()) return 0.0.toFloat()
+    if (imageProxy.width <= 2 || imageProxy.height <= 2) return 0.0.toFloat()
```
Comment (robin-smileid) [inline lib/src/main/java/com/smileidentity/util/Util.kt:129]: why is the size threshold set to 2? would it make more sense to set it to something higher? when would we expect to see a image of such small size?

File: lib/src/main/java/com/smileidentity/viewmodel/document/DocumentCaptureViewModel.kt [hunk 4]
```diff
@@ -384,15 +373,50 @@ class DocumentCaptureViewModel(
...
+                                file = documentFile,
+                                bitmap = bitmap,
+                                desiredAspectRatio = uiState.value.idAspectRatio,
+                            )
+
+                            _uiState.update {
+                                it.copy(
+                                    documentImageToConfirm = processedImage,
+                                    showCaptureInProgress = false,
+                                )
```
Comment (robin-smileid) [inline lib/src/main/java/com/smileidentity/viewmodel/document/DocumentCaptureViewModel.kt:417]: Shouldn't we call https://github.com/smileidentity/android/blob/a710a1ab1d20d53d827c0183ed66ed49758f4c9f/lib/src/main/java/com/smileidentity/viewmodel/document/DocumentCaptureViewModel.kt#L142 here? If we do it like this we also need to set all the metadata as well.
